ext/src/http/client/curl/http_client_curl.cc

```diff
@@ -57,11 +60,85 @@ nostd::shared_ptr<HttpCurlGlobalInitializer> HttpCurlGlobalInitializer::GetInsta
   return shared_initializer;
 }
 
+#ifdef ENABLE_OTLP_COMPRESSION_PREVIEW
+// Original source:
+// https://stackoverflow.com/questions/12398377/is-it-possible-to-have-zlib-read-from-and-write-to-the-same-memory-buffer/12412863#12412863
+int deflateInPlace(z_stream *strm, unsigned char *buf, uint32_t len, uint32_t *max)
+{
+  // must be large enough to hold zlib or gzip header (if any) and one more byte -- 11 works for the
+  // worst case here, but if gzip encoding is used and a deflateSetHeader() call is inserted in this
+  // code after the deflateReset(), then the 11 needs to be increased to accommodate the resulting
+  // gzip header size plus one
+  std::array<unsigned char, 11> temp{};
+
+  // kick start the process with a temporary output buffer -- this allows deflate to consume a large
+  // chunk of input data in order to make room for output data there
+  strm->next_in  = buf;
+  strm->avail_in = len;
+  if (*max < len)
+  {
+    *max = len;
+  }
+  strm->next_out  = temp.data();
+  strm->avail_out = std::min(static_cast<decltype(z_stream::avail_out)>(temp.size()), *max);
+  auto ret        = deflate(strm, Z_FINISH);
+  if (ret == Z_STREAM_ERROR)
+  {
+    return ret;
+  }
+
+  // if we can, copy the temporary output data to the consumed portion of the input buffer, and then
+  // continue to write up to the start of the consumed input for as long as possible
+  auto have = strm->next_out - temp.data();  // number of bytes in temp[]
+  if (have <= (strm->avail_in ? len - strm->avail_in : *max))
```

Comment:
In my understandarding, compare a signed interger with a  unsigned integer will cause a warning. No idea why this warning does not be triggered.

Reply:
For `ENABLE_OTLP_COMPRESSION_PREVIEW` to work, add `-DWITH_OTLP_HTTP_COMPRESSION=ON` to CMake in the maintainer builds.

Reply:
I've not seen compile warnings related to this but. indeed, the type of *have* here is `ptrdiff_t`. At first glance, it does not seem possible that the logic yield a negative number so it could but made unsigned but, given that L96 uses it to advance through the buffer, I think it is more reasonable to keep its intended type and to cast the right hand side of condition in L93 instead.